### PR TITLE
User story three

### PR DESCRIPTION
**Pull request by RBN2208**

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -1,28 +1,80 @@
-// import { v4 as uuidv4 } from 'uuid'
-import { serviceData } from '../../serviceData.json'
+import { v4 as uuidv4 } from 'uuid'
+import { useEffect } from 'react'
```
Comment (macfrei) [inline src/components/App/App.js:2]:
```suggestion
import { useEffect, useState } from 'react'
```

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,28 +1,80 @@
...
+import { useState } from 'react'
```
Comment (macfrei) [inline src/components/App/App.js:3]:
```suggestion

```

File: src/lib/loadLocal.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export default function loadLocal(key) {
```
Comment (macfrei) [inline src/lib/loadLocal.js:1]:
```suggestion
export default function loadFromLocal(key) {
```

File: src/lib/saveLocal.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export default function saveLocal(key, data) {
```
Comment (macfrei) [inline src/lib/saveLocal.js:1]:
```suggestion
export default function saveToLocal(key, data) {
```

File: src/components/App/App.js [hunk 5]
```diff
@@ -1,28 +1,80 @@
...
-          key={id}
-          name={name}
-          costs={costs}
-          onPlus={handlePlus}
-          onMinus={handleMinus}
-        />
-      ))}
-      <Result resultValue={sum} />
-    </AppLayout>
+    <>
```
Comment (macfrei) [inline src/components/App/App.js:23]: No need for the fragments here as `AppLayout` is wrapping around your components.

File: src/components/FormComponents/NewService.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import styled from 'styled-components/macro'
+import Button from '../Button/Button'
+
+export default function NewService({ onSubmit, onAddNewService }) {
+  return (
+    <>
```
Comment (macfrei) [inline src/components/FormComponents/NewService.js:6]: Again, no need for fragments here.

File: src/components/App/App.js [hunk 7]
```diff
@@ -1,28 +1,80 @@
...
+            bgColor={{ name: 'white' }}
+          >
+            New
+          </ButtonNewService>
+          <Result resultValue={sum} />
+          <Delete onClick={() => localStorage.clear()}>clear</Delete>
+        </ButtonBox>
+        {openServiceFrom === 'newService' && (
+          <NewService
+            onSubmit={addNewService}
```
Comment (macfrei) [inline src/components/App/App.js:52]: It is very confusing that you call the function `addNewService` on submit and not `onAddNewService`.

File: src/components/App/App.js [hunk 8]
```diff
@@ -1,28 +1,80 @@
...
+  }
+
+  function addNewService({ name, costs }) {
+    const newService = {
+      id: uuidv4(),
+      name,
+      costs,
+    }
+    setServices([...services, newService])
+  }
```
Comment (macfrei) [inline src/components/App/App.js:76]: Here you can just update the state: `setOpenServiceFrom('home')`. No need for an extra prop/function, I think.

File: src/components/App/App.js [hunk 9]
```diff
@@ -1,28 +1,80 @@
...
+          >
+            New
+          </ButtonNewService>
+          <Result resultValue={sum} />
+          <Delete onClick={() => localStorage.clear()}>clear</Delete>
+        </ButtonBox>
+        {openServiceFrom === 'newService' && (
+          <NewService
+            onSubmit={addNewService}
+            onAddNewService={setOpenServiceFrom}
```
Comment (macfrei) [inline src/components/App/App.js:53]: I would not do this in a separate function but in your submit function `addNewService`.

File: src/components/FormComponents/NewService.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  )
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    const formElements = event.target.elements
+    const data = {
+      name: formElements.service.value,
+      costs: Number(formElements.costs.value) || Number(50),
+    }
+    onAddNewService('home')
```
Comment (macfrei) [inline src/components/FormComponents/NewService.js:39]:
```suggestion
    onAddNewService(data)
```

File: src/components/FormComponents/NewService.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    onSubmit(data)
```
Comment (macfrei) [inline src/components/FormComponents/NewService.js:40]:
```suggestion

```

File: src/components/FormComponents/NewService.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
+import styled from 'styled-components/macro'
+import Button from '../Button/Button'
+
+export default function NewService({ onSubmit, onAddNewService }) {
```
Comment (macfrei) [inline src/components/FormComponents/NewService.js:4]:
```suggestion
export default function NewService({ onAddNewService }) {
```

File: src/components/FormComponents/ServicecardInfo.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+export default function ServicecardInfo({
+  counter,
+  currentCosts,
+  setUsedCosts,
+  onAddingNewCosts,
+  index,
+}) {
+  return (
+    <>
```
Comment (macfrei) [inline src/components/FormComponents/ServicecardInfo.js:12]: Again ;-)

File: src/components/FormComponents/ServicecardInfo.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+      <ServiceInfoForm
+        data-testid="setCostsForm"
+        onSubmit={event => handleSettingNewCosts(event)}
+      >
+        <div>Zeit: {counter} Stunden</div>
+        <div>
+          <label>
+            Stundensatz:
+            <NoSpinnersInput
```
Comment (macfrei) [inline src/components/FormComponents/ServicecardInfo.js:21]: This name is not really descriptive, how about:
```suggestion
            <CostInput
```

**Review by macfrei — COMMENTED** [review]
Nice! I just found a few things which you might want to consider! Keep up the good work :)

File: src/components/App/App.js (hunk 1 above)
Comment (RBN2208) [reply src/components/App/App.js]: Hey, thanks for your review! Here is a link to the commit with the updates!
https://github.com/RBN2208/capstone-project/pull/9/commits/6b023952de91cd64a5e7cf99561fa49d5162906c

**Pull request merged**